Userland/Libraries/LibJS/Runtime/ArrayBuffer.cpp

```diff
@@ -201,17 +219,20 @@ ThrowCompletionOr<ArrayBuffer*> array_buffer_copy_and_detach(VM& vm, ArrayBuffer
     if (array_buffer.is_detached())
         return vm.throw_completion<TypeError>(ErrorType::DetachedArrayBuffer);
 
-    // FIXME: 6. If preserveResizability is preserve-resizability and IsResizableArrayBuffer(arrayBuffer) is true, then
+    // 6. If preserveResizability is preserve-resizability and IsResizableArrayBuffer(arrayBuffer) is true, then
     // a. Let newMaxByteLength be arrayBuffer.[[ArrayBufferMaxByteLength]].
     // 7. Else,
     // a. Let newMaxByteLength be empty.
+    auto new_max_byte_length = preserve_resizability == PreserveResizability::PreserveResizability && array_buffer.is_resizable()
+        ? array_buffer.max_byte_length()
+        : Optional<size_t> {};
 
     // 8. If arrayBuffer.[[ArrayBufferDetachKey]] is not undefined, throw a TypeError exception.
     if (!array_buffer.detach_key().is_undefined())
         return vm.throw_completion<TypeError>(ErrorType::DetachKeyMismatch, array_buffer.detach_key(), js_undefined());
 
     // 9. Let newBuffer be ? AllocateArrayBuffer(%ArrayBuffer%, newByteLength, FIXME: newMaxByteLength).
```

Comment:
// 9. Let newBuffer be ? AllocateArrayBuffer(%ArrayBuffer%, newByteLength, newMaxByteLength).